ignite/metrics/metric.py

```diff
@@ -168,11 +175,10 @@ def attach(self, engine: Engine, name: str) -> None:
 
             assert metric.is_attached(engine)
         """
-        engine.add_event_handler(Events.EPOCH_COMPLETED, self.completed, name)
-        if not engine.has_event_handler(self.started, Events.EPOCH_STARTED):
-            engine.add_event_handler(Events.EPOCH_STARTED, self.started)
-        if not engine.has_event_handler(self.iteration_completed, Events.ITERATION_COMPLETED):
-            engine.add_event_handler(Events.ITERATION_COMPLETED, self.iteration_completed)
+        if name is not None:
```

Comment:
Why do not use Attachable here ?

Reply:
This was again to do with the way the `MetricsLambda` class worked, the partial attaching mechanism is what is supported by this way of doing things.